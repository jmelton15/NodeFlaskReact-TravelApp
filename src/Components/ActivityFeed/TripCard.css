
.TripCard-Card{
    border-radius: 20px !important;
}


.TripCard-TripDetails{
    overflow-y:scroll;
    max-height: 300px ;
}

#tripcard-tripimg {
    height:250px;
    border:2px solid rgb(108, 117, 125);
    box-shadow: 5px 5px 30px grey;
}

#tripcard-useravatar {
   height:70px;
   width:70px;
   /* border-radius: 50px; */
   margin-top: 10px;
   margin-left: 20px;
   border:2px solid rgb(108, 117, 125);
   box-shadow: 5px 5px 30px grey;
}

#tripcard-username {
    color: #E97451;
    font-family: "AestheticViolet";
    /* font-weight: bold; */
    text-decoration: underline dotted black;
    position:absolute;
    right:42%;
    margin-top: 10px;
}

#tripcard-likecountbadge {
    color:black;
}