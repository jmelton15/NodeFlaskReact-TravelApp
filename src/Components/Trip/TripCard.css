.TripCard-Card {
    
}

.TripCard-Card{
    border-radius: 20px !important;
}

#tripcard-tripimg {
    border:2px solid rgb(108, 117, 125);
    box-shadow: 5px 5px 30px grey;
}

#tripcard-useravatar {
   height:80px;
   width:80px;
   border-radius: 50px;
   margin-top: 10px;
   margin-left: 20px;
   border:3px solid rgb(108, 117, 125);
}

#tripcard-username {
    text-decoration: underline;
}