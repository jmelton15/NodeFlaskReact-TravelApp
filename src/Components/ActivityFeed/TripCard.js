import React from 'react'
import {
    Card, CardImg, CardText, CardBody,
    CardTitle, CardSubtitle, Button, Row,Col, Container, Badge
  } from 'reactstrap';
import {FontAwesomeIcon} from "@fortawesome/react-fontawesome";
import {faThumbsUp as tuSolid,faThumbsDown as tdSolid} from "@fortawesome/free-solid-svg-icons";
import {faThumbsUp as tuReg,faThumbsDown as tdReg} from "@fortawesome/free-regular-svg-icons";
import {useState,useEffect} from "react";
import {NodeApi} from "../../APIRequests/nodeApi";
import {hasValue} from "../../helpers/helpers";
import "./TripCard.css";
import GetScreenWidth from '../../helpers/GetScreenWidth';
import {v4 as uuid} from "uuid";
import testAvatar from "../../Images/default_avatar.jpg"

function TripCard({user,trip,token}) {
    const [screenWidth] = GetScreenWidth();
    const [hasLiked,setHasLiked] = useState(false);
    const [likes,setLikes] = useState(trip.like_count);
    
    const toggleHasLiked = () => {
        setHasLiked(!hasLiked);
    }
    
    const likeTrip = async (tripId) => {
        const likedTrip = await NodeApi.likeTrip(user.user_id,tripId,token);
        toggleHasLiked();
        setLikes(like => like = like + 1)
    }
    let hasLikes = user.liked_trips !== [] ? true : false;

    useEffect(() => {
        const checkIfHasLiked = () => {
            if(hasValue(user.liked_trips,"trip_id",trip.trip_id)) return true;
            else {
                return false;
            }
        }
        if(hasLikes) {
            let liked = checkIfHasLiked();
            setHasLiked(liked);
        }
    },[hasLikes,trip.trip_id,user.liked_trips]);
    
    
    const unlikeTrip = async (tripId) => {
        const unlikedTrip = await NodeApi.unlikeTrip(user.user_id,tripId,token);
        toggleHasLiked();
        setLikes(like => like = like - 1)
    }


    console.log(JSON.parse(trip.marker_data));
    return (
        <div>
            <Card className="TripCard-Card">
            <div className="d-flex align-items-center">
                <CardImg  id="tripcard-useravatar" src={testAvatar} alt="Profile Avatar" />
                <CardTitle tag="h2" className="text-center" id="tripcard-username">{trip.username.toUpperCase()}</CardTitle>
            </div>
            <CardBody className="TripCard-CardBody">
                <Container fluid className="TripCard-AllRowsContainer">
                {screenWidth <= 420 && 
                    <>
                    <Row>
                        <Col>
                            <div className="TripCard-TripDetails">
                                <CardSubtitle tag="h6" className="mb-2" id="tripcard-startAndStopHeader">
                                    Trip From {trip.start_point.toUpperCase()} To {trip.end_point.toUpperCase()}
                                </CardSubtitle>
                                {trip.marker_data && Object.keys(JSON.parse(trip.marker_data)).map((place) => {
                                    return(<div>
                                        <div><h6 key={uuid()}><u>{place}</u></h6></div>
                                            {JSON.parse(trip.marker_data)[place].map((location) => {
                                            return <div><ul>
                                                    <li key={uuid()} id="traveljournal-waypointlist">
                                                        <a key={uuid()} href={location.web_url}target="_blank" rel="noopener noreferrer" id="travlejournal-placelink">{location.name}</a>
                                                    </li>
                                                </ul> </div>
                                            })}
                                        </div>
                                    )}
                                )} 
                            </div>
                        </Col>
                    </Row>
                    <Row>
                        <Col>
                            <div className="TripCard-TripImageContainer">
                                <img id="tripcard-tripimg" width="100%" src={trip.photo.img_url} alt="Trip" />
                            </div> 
                        </Col>
                    </Row>
                    </>
                }
                   {screenWidth > 420 && <Row>
                        <Col >
                            <div className="TripCard-TripDetails">
                                <CardSubtitle tag="h6" className="mb-2 text-muted">
                                    Trip From {trip.start_point.toUpperCase()} To {trip.end_point.toUpperCase()}
                                </CardSubtitle>
                                {trip.marker_data && Object.keys(JSON.parse(trip.marker_data)).map((place) => {
                                    return(<div>
                                        <div><h6><u>{place}</u></h6></div>
                                            {JSON.parse(trip.marker_data)[place].map((location) => {
                                            return <div><ul>
                                                    <li key={uuid()} id="traveljournal-waypointlist">
                                                        <a href={location.web_url}target="_blank" rel="noopener noreferrer" id="travlejournal-placelink">{location.name}</a>
                                                    </li>
                                                </ul> </div>
                                            })}
                                        </div>
                                    )}
                                )} 
                            </div>
                        </Col>
                        <Col className="d-flex align-items-center">
                            <div className="TripCard-TripImageContainer">
                                <img id="tripcard-tripimg" width="100%" src={trip.photo.img_url} alt="Trip" />
                            </div> 
                        </Col>
                    </Row> }   
                </Container>
                <div className="TripCard-LikeContainer mt-3">
                    <FontAwesomeIcon 
                        data-testid="likeBtn"
                        icon={hasLiked ? tuSolid : tuReg}
                        size="lg" 
                        className="me-3"
                        onClick={() => {
                            hasLiked ? unlikeTrip(trip.trip_id) : likeTrip(trip.trip_id)
                        }
                        }
                    />
                    <Badge id="tripcard-likecountbadge" data-testid="likeCount">{likes}</Badge>
                </div>
            </CardBody>
            
            </Card>
        </div>
    )
}


export default TripCard;
