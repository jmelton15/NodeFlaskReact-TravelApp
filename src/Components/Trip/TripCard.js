import React from 'react'
import {
    Card, CardImg, CardText, CardBody,
    CardTitle, CardSubtitle, Button, Row,Col, Container
  } from 'reactstrap';
import {FontAwesomeIcon} from "@fortawesome/react-fontawesome";
import {faThumbsUp as tuSolid,faThumbsDown as tdSolid} from "@fortawesome/free-solid-svg-icons";
import {faThumbsUp as tuReg,faThumbsDown as tdReg} from "@fortawesome/free-regular-svg-icons";
import {useState,useEffect} from "react";
import {NodeApi} from "../../APIRequests/nodeApi";
import {hasValue} from "../../helpers/helpers";
import "./TripCard.css";


function TripCard({user,trip,token}) {
    const [hasLiked,setHasLiked] = useState(false);
    
    const toggleHasLiked = () => {
        setHasLiked(!hasLiked);
    }
    
    const likeTrip = async (tripId) => {
        const likedTrip = await NodeApi.likeTrip(user.user_id,tripId,token);
        toggleHasLiked();
    }
    let hasLikes = user.liked_trips !== [] ? true : false;

    useEffect(() => {
        const checkIfHasLiked = () => {
            if(hasValue(user.liked_trips,"trip_id",trip.trip_id)) return true;
            else {
                return false;
            }
        }
        if(hasLikes) {
            let liked = checkIfHasLiked();
            setHasLiked(liked);
        }
    },[hasLikes,trip.trip_id,user.liked_trips]);
    
    
    const unlikeTrip = async (tripId) => {
        const unlikedTrip = await NodeApi.unlikeTrip(user.user_id,tripId,token);
        toggleHasLiked();
    }

    return (
        <div>
            <Card className="TripCard-Card">
            <CardImg  id="tripcard-useravatar" src={trip.avatar_pic_url} alt="Profile Avatar" />
            <CardBody className="TripCard-CardBody">
                <Container fluid>
                    <Row>
                        <Col>
                            <CardTitle tag="h5" className="text-center" id="tripcard-username">{trip.username.toUpperCase()}</CardTitle>
                            <CardSubtitle tag="h6" className="mb-2 text-muted">
                                Trip From {trip.start_point.toUpperCase()} To {trip.end_point.toUpperCase()}
                            </CardSubtitle>
                            {trip.waypoint_names && 
                            <ul className="TripCard-WaypointsList">
                                {trip.waypoint_names.map((name) => {
                                    return <li>{name}</li>
                                })}
                            </ul>
                            }
                        </Col>
                        <Col className="d-flex align-items-center">
                            <div className="TripCard-TripImageContainer">
                                <img id="tripcard-tripimg" width="100%" src={trip.photo} alt="Trip" />
                            </div> 
                        </Col>
                    </Row>    
                </Container>
                <div className="TripCard-LikeContainer mt-3">
                    <FontAwesomeIcon 
                        icon={hasLiked ? tuSolid : tuReg}
                        size="lg" 
                        className="me-3"
                        onClick={() => {
                            hasLiked ? unlikeTrip(trip.trip_id) : likeTrip(trip.trip_id)
                        }
                        }
                    />
                </div>
            </CardBody>
            
            </Card>
        </div>
    )
}


export default TripCard;
