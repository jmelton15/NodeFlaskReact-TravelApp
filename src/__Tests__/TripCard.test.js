import {render } from '@testing-library/react';
import {MemoryRouter} from "react-router-dom";
import {v4 as uuid} from "uuid";
import {user,trip} from "./TestHelperObjects";
import TripCard from "../Components/ActivityFeed/TripCard";

// simple smoke test for TripCard component
it("renders a TripCard component without crashing", function() {
    render(<TripCard user={user} trip={trip} token={true} />);
});

// simple snapshot test for TripCard Component
it("should match snapshot for TripCard", function() {
    const {asFragment} = render (<TripCard user={user} trip={trip} token={true} />);
    expect(asFragment()).toMatchSnapshot();
});


