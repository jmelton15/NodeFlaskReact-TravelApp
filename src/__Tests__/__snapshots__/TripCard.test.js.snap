// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot for TripCard 1`] = `
<DocumentFragment>
  <div>
    <div
      class="TripCard-Card card"
    >
      <div
        class="d-flex align-items-center"
      >
        <img
          alt="Profile Avatar"
          class="card-img"
          id="tripcard-useravatar"
          src="some pic route"
        />
        <h2
          class="text-center card-title"
          id="tripcard-username"
        >
          TEST1
        </h2>
      </div>
      <div
        class="TripCard-CardBody card-body"
      >
        <div
          class="TripCard-AllRowsContainer container-fluid"
        >
          <div
            class="row"
          >
            <div
              class="col"
            >
              <div
                class="TripCard-TripDetails"
              >
                <h6
                  class="mb-2 text-muted card-subtitle"
                >
                  Trip From GRAND RAPIDS To MACKINAW CITY
                </h6>
                <ul
                  class="TripCard-WaypointsList"
                >
                  <li>
                    The Leonard At Logan House
                  </li>
                  <li>
                    Tru By Hilton Grand Rapids Airport
                  </li>
                  <li>
                    Whispering Waters B & B
                  </li>
                </ul>
              </div>
            </div>
            <div
              class="d-flex align-items-center col"
            >
              <div
                class="TripCard-TripImageContainer"
              >
                <img
                  alt="Trip"
                  id="tripcard-tripimg"
                  src="https://maps.googleapis.com/maps/api/place/photo?maxwidth=1600&maxheight=300&photoreference=Aap_uEAcABGaNC12R6jbBdUFgnBZElfHDYCTtD6Z2rDPRQH-ZBmNfBqA-SfzXVDHeNJpvB9s_nNKkaH4cM-QqWkzGiX8Cbwl5hKfc0H0lCGMkP5fy151S-twKvmLYlWv6khDlWB-46Ma1tmGn0e2K_ksg6GCLHSIGI6tiGW1AU-1eL0QFJ1o&key=AIzaSyCjXBkeGKHmGE6k4iXPddI-2ye-OXZ8k30"
                  width="100%"
                />
              </div>
            </div>
          </div>
        </div>
        <div
          class="TripCard-LikeContainer mt-3"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-thumbs-up fa-w-16 fa-lg me-3"
            data-icon="thumbs-up"
            data-prefix="far"
            data-testid="likeBtn"
            focusable="false"
            role="img"
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M466.27 286.69C475.04 271.84 480 256 480 236.85c0-44.015-37.218-85.58-85.82-85.58H357.7c4.92-12.81 8.85-28.13 8.85-46.54C366.55 31.936 328.86 0 271.28 0c-61.607 0-58.093 94.933-71.76 108.6-22.747 22.747-49.615 66.447-68.76 83.4H32c-17.673 0-32 14.327-32 32v240c0 17.673 14.327 32 32 32h64c14.893 0 27.408-10.174 30.978-23.95 44.509 1.001 75.06 39.94 177.802 39.94 7.22 0 15.22.01 22.22.01 77.117 0 111.986-39.423 112.94-95.33 13.319-18.425 20.299-43.122 17.34-66.99 9.854-18.452 13.664-40.343 8.99-62.99zm-61.75 53.83c12.56 21.13 1.26 49.41-13.94 57.57 7.7 48.78-17.608 65.9-53.12 65.9h-37.82c-71.639 0-118.029-37.82-171.64-37.82V240h10.92c28.36 0 67.98-70.89 94.54-97.46 28.36-28.36 18.91-75.63 37.82-94.54 47.27 0 47.27 32.98 47.27 56.73 0 39.17-28.36 56.72-28.36 94.54h103.99c21.11 0 37.73 18.91 37.82 37.82.09 18.9-12.82 37.81-22.27 37.81 13.489 14.555 16.371 45.236-5.21 65.62zM88 432c0 13.255-10.745 24-24 24s-24-10.745-24-24 10.745-24 24-24 24 10.745 24 24z"
              fill="currentColor"
            />
          </svg>
          <span
            class="badge badge-secondary"
            data-testid="likeCount"
            id="tripcard-likecountbadge"
          >
            0
          </span>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
